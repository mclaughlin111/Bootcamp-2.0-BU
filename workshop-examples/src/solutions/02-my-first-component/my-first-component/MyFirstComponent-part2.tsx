type MyFirstComponentProps = {
  message: string;
};

export const MyFirstComponent = ({ message }: MyFirstComponentProps) => (
  <h2>{message}</h2>
);

export function MyFirstComponentFunction({ message }: MyFirstComponentProps) {
  return <h2>{message}</h2>;
}
