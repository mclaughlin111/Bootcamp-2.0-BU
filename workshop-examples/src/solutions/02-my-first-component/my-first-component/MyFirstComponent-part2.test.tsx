import { render, screen } from "@testing-library/react";
import { MyFirstComponent } from "./MyFirstComponent-part2";
import { MyFirstComponentFunction } from "./MyFirstComponent-part2";

describe("non snapshot testing", () => {
  // 2 tests, less efficient

  test("with hello world message", () => {
    const testMessage = "hello world!";

    render(<MyFirstComponent message={testMessage} />);

    const headerElement = screen.getByRole("heading", {
      level: 2,
      name: testMessage,
    });
    expect(headerElement).toBeInTheDocument();
  });

  test("with different message", () => {
    const testMessage = "different!";

    render(<MyFirstComponent message={testMessage} />);

    const headerElement = screen.getByRole("heading", {
      level: 2,
      name: testMessage,
    });
    expect(headerElement).toBeInTheDocument();
  });

  // The same thing in one test

  test.each(["hello world!", "different!"])(
    "test with multiple messages (message=%s)",
    (testMessage) => {
      render(<MyFirstComponentFunction message={testMessage} />);

      const headerElement = screen.getByRole("heading", {
        level: 2,
        name: testMessage,
      });
      expect(headerElement).toBeInTheDocument();
    }
  );
});

describe("snapshot testing", () => {
  test.each(["hello world!", "different!"])(
    "test with multiple messages (message=%s)",
    (testMessage) => {
      const { container } = render(
        <MyFirstComponentFunction message={testMessage} />
      );

      expect(container).toMatchSnapshot();
    }
  );
});
