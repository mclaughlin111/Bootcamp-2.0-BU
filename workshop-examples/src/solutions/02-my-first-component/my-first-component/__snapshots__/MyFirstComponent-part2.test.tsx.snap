// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing test with multiple messages (message=different!) 1`] = `
<div>
  <h2>
    different!
  </h2>
</div>
`;

exports[`snapshot testing test with multiple messages (message=hello world!) 1`] = `
<div>
  <h2>
    hello world!
  </h2>
</div>
`;
